### PR TITLE
new resource: azurerm_eventhub_namespace_schema_group

### PR DESCRIPTION
**Pull request by xiaxyi**
adding support for eventhub namespace schema group

acc tests:
--- PASS: TestAccEventHubNamespaceSchemaRegistry_basic (440.94s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/eventhub      441.626s

### CODE REVIEW

File: internal/services/eventhub/eventhub_namespace_schema_registry_resource.go
Comment (katbyte) [inline internal/services/eventhub/eventhub_namespace_schema_registry_resource.go:74]: is this PR a work in progress?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/eventhub/eventhub_namespace_schema_registry_resource.go
Comment (xiaxyi) [reply internal/services/eventhub/eventhub_namespace_schema_registry_resource.go:74]: @katbyte The group property implementation is blocked by SDK issue, in short words, the CRUD operation of this property cannot be achieved by the current type in SDK: the deletion requires a pointer while it's a string in currently released SDK. I'll add it once we work the blocker out.

**Comment by xiaxyi** [comment]
@katbyte Thanks for the comment, I left the feedback inline. Feel free to let me know if you have any concerns.

File: internal/services/eventhub/eventhub_namespace_schema_registry_resource.go
Comment (katbyte) [inline internal/services/eventhub/eventhub_namespace_schema_registry_resource.go:53]: does type make any sense here?
```suggestion
			"schema_compatibility": {
```

Comment (xiaxyi) [reply internal/services/eventhub/eventhub_namespace_schema_registry_resource.go:53]: Yes, resource is functioning just fine without this property. I'll modify the naming

File: internal/services/eventhub/eventhub_namespace_schema_registry_resource.go
Comment (xiaxyi) [reply internal/services/eventhub/eventhub_namespace_schema_registry_resource.go:75]: @katbyte I think we better keep the property suffix here as this `schemaGroupProperty` is letting users to actually set the property of the schema group:
[SchemaGroup](https://docs.microsoft.com/en-us/rest/api/eventhub/stable/schema-registry/create-or-update?tabs=HTTP#schemagroup)

**Comment by xiaxyi** [comment]
Thanks @katbyte, updated the pr

File: internal/services/eventhub/eventhub_namespace_schema_registry_resource.go
Comment (katbyte) [reply internal/services/eventhub/eventhub_namespace_schema_registry_resource.go:74]: But does it work as a resource without it? ie does it function for the user without specifying the property? as it seems the point of the resource is to set that property

File: internal/services/eventhub/eventhub_namespace_schema_registry_resource.go
Comment (xiaxyi) [reply internal/services/eventhub/eventhub_namespace_schema_registry_resource.go]: The resource is stable without this `groupProperty`. It's not a required one when creating the schema group in the portal:
image

based on my understanding, it should have the same funtion as `Tag` in the resource: user to speficy the key and the corresponding value.

File: internal/services/eventhub/eventhub_namespace_schema_registry_resource.go
Comment (xiaxyi) [reply internal/services/eventhub/eventhub_namespace_schema_registry_resource.go:74]: The resource is stable without this `groupProperty`. It's not a required one when creating the schema group in the portal:
image

based on my understanding, it should have the same funtion as `Tag` in the resource: user to speficy the key and the corresponding value.

**Comment by xiaxyi** [comment]
Thanks @katbyte for the comments, responded inline

File: internal/services/eventhub/eventhub_namespace_schema_registry_resource.go
Comment (katbyte) [reply internal/services/eventhub/eventhub_namespace_schema_registry_resource.go:75]: i mean this isn't even implemented, so can we remove the comment for now and we can revisit this when it is ready to be implemented.

**Review by katbyte — APPROVED** [review]
Thanks @xiaxyi - aside from the commented out property this looks good - lets remove the comments for now and leave that for when it is implemented.

LGTM once the comments are removed and tests pass

**Comment by xiaxyi** [comment]
Thanks @katbyte, I removed the comments, let me know if it looks cool to you

**Review by katbyte — APPROVED** [review]
Thanks @xiaxyi - LGTM 🎩

**Pull request merged**
